### PR TITLE
Iss450 fix vertical windspeed column issue

### PR DESCRIPTION
**Pull request by olivia-bentley**

### CODE REVIEW

File: brightwind/load/station.py [hunk 1]
```diff
@@ -1071,9 +1071,10 @@ def __get_names(self, measurement_type_id=None):
         :param measurement_type_id: The measurement_type_id to filter for the names.
         :type measurement_type_id:  str or None
         :return:                    The list of names.
-        :rtype:                     list(str)
+        :rtype:                     str | list[str]
```
Comment (BiancaMorandi) [inline brightwind/load/station.py:1074]: I don't get why you made this update. The output of the function before and after your fix is a list(str), it will never happen that it is a str as `names` is defined as a list on function e.g. `names = []`. Did you mean to update the input 'measurement_type_id' type?

File: brightwind/load/station.py [hunk 2]
```diff
@@ -1071,9 +1071,10 @@ def __get_names(self, measurement_type_id=None):
...
         """
         names = []
+        measurement_type_id = measurement_type_id if isinstance(measurement_type_id, list) else [measurement_type_id]
```
Comment (BiancaMorandi) [inline brightwind/load/station.py:1077]: probably this is not the best way to fix this bug, this fix has added an additional bug as doesn't handle cases where `measurement_type_id=None` e.g if I run `mm1.measurements._Measurements__get_names(measurement_type_id=None)` I get an empty list, but I should get a list of all measurement names instead . Can you fix this?

image

Based on what stated on docstring, it looks like this function was initially made for working with only one measurement_type (str) or all (None) and this is also the behaviour of other functions calling this (e.g `get_heights`). @stephenholleran do you want this function to be updated to work with `measurement_type_id ` both str and list(str) e.g pull out names also for multiple measurement types? 

If we keep the current behaviour accepting only `measurement_type_id `as as str then I would suggest to fix this bug editing the line highlighted below to be as below and raise an error if `measurement_type_id ` is not str or  None

`if meas_type is not None and meas_type == measurement_type_id`

image

File: tests/test_station.py [hunk 3]
```diff
@@ -66,3 +66,17 @@ def test_get_table():
         if value is not '-':
             assert value in mm2.measurements.properties[4].values()
 
+
+def test_get_names():
```
Comment (BiancaMorandi) [inline tests/test_station.py:70]: after updating `names `function as for Stephen's request then you can probably change this test to be for `names` and maybe leave few tests for the private `_get_names` function just to be safe

can you add tests also for when `measurement_type_id=None`?

File: tests/test_station.py [hunk 4]
```diff
@@ -66,3 +66,17 @@ def test_get_table():
...
+    mm1 = bw.MeasurementStation(bw.demo_datasets.floating_lidar_iea43_wra_data_model_v1_2)
+    measurement_type_id1= 'vertical_wind_speed'
+    names1 = mm1.measurements._Measurements__get_names(measurement_type_id1)
+    assert names1 == ['VSpd_80']
+    
+    measurement_type_id2= 'wind_speed'
+    names2 = mm1.measurements._Measurements__get_names(measurement_type_id2)
+    assert names2 == ['Spd_80']
+    
+    measurement_type_id3= ['wind_speed', 'vertical_wind_speed']
```
Comment (BiancaMorandi) [inline tests/test_station.py:80]: this test is needed only if we want the function to accept list(str) see question I have asked Stephen above

**Review by BiancaMorandi — CHANGES_REQUESTED** [review]
@olivia-bentley see my comments

File: brightwind/load/station.py (hunk 2 above)
Comment (stephenholleran) [reply brightwind/load/station.py:1077]: > @stephenholleran do you want this function to be updated to work with measurement_type_id  both str and list(str) e.g pull out names also for multiple measurement types?

Not the plan but might be useful. I would do that as an enhancement in another issue. I think we should just fix the bug here by using the "==" as opposed to "in" so the user would need to type out the full measurement type name.

Comment (BiancaMorandi) [reply brightwind/load/station.py:1077]: ok, @olivia-bentley can you make updates according to Stephen's comment?

File: tests/test_station.py (hunk 3 above)
Comment (BiancaMorandi) [reply tests/test_station.py:70]: we decided to keep `names` function as it is and change `get_names` to be public, so ok to have tests for get_names

**Comment by BiancaMorandi** [comment]
@stephenholleran merge request for this issue is ready for your review/approval. We have made changes as reported on comments on issue #450

**Review by stephenholleran — APPROVED** [review]
